LL-5947 Change the TC/PP screen into a Legal screen

jgiamporcaro-ledger opened the pull request:
<!-- Description of what the PR does go here... screenshot might be good if appropriate -->
Rework terms and conditions from a modal that load static terms and conditions to a dedicated page that redirects the user to the terms and conditions webpage.

<details>
  <summary>Onboarding step screenshots 📸 </summary>
<img width="1005" alt="Screenshot 2021-07-08 at 13 22 53" src="https://user-images.githubusercontent.com/33158502/124921652-e910ae00-dff8-11eb-83c2-fe0e5827228e.png">
<img width="1003" alt="Screenshot 2021-07-08 at 13 23 04" src="https://user-images.githubusercontent.com/33158502/124921656-e9a94480-dff8-11eb-9f88-50cefcbf70ca.png">
</details>

<details>
  <summary>Settings screenshots 📸 </summary>
<img width="1012" alt="Screenshot 2021-07-08 at 13 23 45" src="https://user-images.githubusercontent.com/33158502/124921722-fb8ae780-dff8-11eb-9569-b3eef2a37b22.png">
<img width="1017" alt="Screenshot 2021-07-08 at 13 23 51" src="https://user-images.githubusercontent.com/33158502/124921728-fcbc1480-dff8-11eb-9e8c-bd44d0f81a30.png">

</details>

### Type

Feature

### Context

[LL-5947]

### Parts of the app affected / Test plan

This pull request adds a new onboarding step, step 2. From now, a new screen is displayed between the first onboarding screen and the screen to pick a device.
In addition, once logged in, the menu in the item for displaying terms (from settings/about) no longer displays a modal but redirects the user to a web page.

<!-- Which part of the app is affected? What to do to test it, any special thing to do? -->


[LL-5947]: https://ledgerhq.atlassian.net/browse/LL-5947

jgiamporcaro-ledger commented on src/renderer/terms.js:
```diff
@@ -31,38 +24,26 @@ export function acceptLendingTerms() {
 
-export const usePrivacyUrl = () => {
-  const [privacyUrl, setPrivacyUrl] = useState(urls.privacyPolicy.en);
+// TODO: Merge useTermsUrl and usePrivacyUrl to create generic useDynamicUrl
```
Will be happy to take few seconds to do that.
I'm thinking about developing a generic hook (`useDynamicUrl`) that accepts a translationKey param and return the correct URL if found. It will replace `useTermsUrl` and `usePrivacyUrl` which are redundant. 
Tell me if I have to include this modification to this pull request 👌

LFBarreto replied:
Sure go for it. just make sure to search on the app all the instances wher this is called :+1:

jgiamporcaro-ledger commented:
You'll see that buttons used to redirect users to webpages from the new onboarding screen aren't pixel perfect. That because I used existing components to display them, as discussed with my team.
Styles from the components I used have some hardcoded styles which force `justify-content: flex-start` rule. To move the svg icon to the right of the button, we've to update this component by adding new props or create new one.

As I understand, new rebranding is coming so the entire app will be redesign. Tell me if this is a problem.

LFBarreto commented on static/i18n/ar/app.json:
```diff
@@ -2908,11 +2908,10 @@
-    "switchLabelTerms": "<0>لقد قرأت ووافقت على </0><1>شروط الاستخدام</1>",
-    "switchLabelPrivacyPolicy": "<0>لقد قرأت ووافقت على</0> <1>سياسة الخصوصية</1>",
-    "switchLabelLoss": "أفهم أنني إذا فقدت عبارة الاسترداد هذه، فلن أتمكن من الوصول إلى أصولي المشفرة في حالة فقداني القدرة على الوصول إلى جهاز نانو.",
+    "title": "Legal",
```
maybe remove these changes to other languages than english :sweat_smile: 
You're invalidating translations here.
As a rule of thumbs you should never edit translation files unless its the `en` one to edit or add onto it

jgiamporcaro-ledger replied:
Got it!

jgiamporcaro-ledger replied:
Done -> [fee0525542d7fcc174bf95556bb9d7475605c4c0](https://github.com/LedgerHQ/ledger-live-desktop/pull/3967/commits/fee0525542d7fcc174bf95556bb9d7475605c4c0)

LFBarreto reviewed (changes requested):
I would revert changes on the translation files.
Also i think we have a way to check the terms again from the settings of the app. make sure it was not broken by the removal of MODAL_TERMS :+1:

jgiamporcaro-ledger commented:
@LFBarreto I pushed some updates based on your comments.

- I removed translations for non-en languages [fee052](https://github.com/LedgerHQ/ledger-live-desktop/pull/3967/commits/fee0525542d7fcc174bf95556bb9d7475605c4c0)
- I merged `useTerms`/`usePrivacyUrl` into one generic new hook called `useDynamicUrl` [4d394a](https://github.com/LedgerHQ/ledger-live-desktop/pull/3967/commits/4d394a35c992c66514eef3c86377aeb66bd37e61)

> Also i think we have a way to check the terms again from the settings of the app. make sure it was not broken by the removal of MODAL_TERMS 👍

Yep, I can confirm that I thought about that and tested it. If you want to see the change, I added screenshots of this part of the app in my initial message [here](https://github.com/LedgerHQ/ledger-live-desktop/pull/3967#issue-685972534). Basically, I applied the same UI logic from the "Privacy Policy" component to the "Terms of use" component, no modal is needed anymore, users are automatically redirected to the correct webpage.

LFBarreto reviewed (approved):
:octocat:

LFBarreto reviewed (approved):
:rocket:

LFBarreto reviewed (changes requested):
update screenshots and we are good to go :+1:

jgiamporcaro-ledger commented:
**Some updates here**

This task highlighted a CI issue. Some snapshots failed as expected but weren’t shared for download by our CI bot. To be able to update changed snapshots I have to run end-to-end tests locally but the process to run it seems broken. Passed some time on Monday to fix it without any success. I asked for some help from Nabil who is going to take some time this week to deep dive into the issue. Once the issue fixes, the PR would be ready to merge.

jgiamporcaro-ledger commented:
Failed tests are unrelated to my changes.

jgiamporcaro-ledger commented:
Rebased develop branch resolved recent conflicts.

jgiamporcaro-ledger commented:
What breaks the tests isn't related to my changes.

ychen-ledger reviewed (approved):
Validated with: 

- MacOs Big Sur 11.4
- Ledger live 2.32.4
